test/system/notification/tabs_badges_test.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+require 'application_system_test_case'
+
+class Notification::TabsBadgesTest < ApplicationSystemTestCase
+  setup do
+    Notification.create!(
+      message: 'お知らせ',
+      kind: 5,
+      link: '/notifications/1',
+      user: users(:sotugyou),
+      sender: users(:komagata),
+      read: false
+    )
+
+    Notification.create!(
+      message: 'メンション',
+      kind: 2,
+      link: '/notifications/2',
+      user: users(:sotugyou),
+      sender: users(:komagata),
+      read: false
+    )
+
+    Notification.create!(
+      message: 'コメント',
+      kind: 0,
+      link: '/notifications/3',
+      user: users(:sotugyou),
+      sender: users(:komagata),
+      read: false
+    )
+
+    Notification.create!(
+      message: '提出物の確認',
+      kind: 1,
+      link: '/notifications/4',
+      user: users(:sotugyou),
+      sender: users(:komagata),
+      read: false
+    )
+
+    Notification.create!(
+      message: 'watch中',
+      kind: 8,
+      link: '/notifications/5',
+      user: users(:sotugyou),
+      sender: users(:komagata),
+      read: false
+    )
+    Notification.create!(
+      message: 'フォロー中',
+      kind: 13,
+      link: '/notifications/6',
+      user: users(:sotugyou),
+      sender: users(:komagata),
+      read: false
+    )
+  end
+  test 'unread badges are displayed' do
+    visit_with_auth '/notifications', 'sotugyou'
+
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(1) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(2) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(3) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(4) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(5) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(6) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
+    assert_selector '#body > div.wrapper > main > div > div > ul > li:nth-child(7) > a > div' do
+      assert_selector 'div.page-tabs__item-count.a-notification-count'
+    end
```

Comment:
```suggestion
    within all('.page-tabs__item')[0] do
      assert_selector '.page-tabs__item-count'
    end

    within all('.page-tabs__item')[1] do
      assert_selector '.page-tabs__item-count'
    end

    within all('.page-tabs__item')[2] do
      assert_selector '.page-tabs__item-count'
    end

    within all('.page-tabs__item')[3] do
      assert_selector '.page-tabs__item-count'
    end

    within all('.page-tabs__item')[4] do
      assert_selector '.page-tabs__item-count'
    end

    within all('.page-tabs__item')[5] do
      assert_selector '.page-tabs__item-count'
    end

    within all('.page-tabs__item')[6] do
      assert_selector '.page-tabs__item-count'
    end
```

とするのはどうでしょう？[伊藤さんのこの記事](https://qiita.com/jnchito/items/7bd131d99c28f21834dd)を参考に書いてみました。
`unread badges are not displayed`の方も同様にして書けます。